test: include component in tap output

bnoordhuis opened the pull request:
Print test name as (for example) "parallel/test-assert".  Tests that are
scraped from the addons documentation are all named test.js, making it
hard to decipher what test is running when only the filename is printed.

Fixes: #6651 

CI: https://ci.nodejs.org/job/node-test-pull-request/2547/

Fishrock123 commented:
idea sgtm but won't review python XD

gibm commented:
@bnoordhuis Does the node-test-commit not use the new tools/test.py? Seems to still be test.js for the addons.

https://ci.nodejs.org/job/node-test-commit-linux/nodes=ubuntu1404-64/3282/console

bnoordhuis commented:
It's possible the buildbots aren't actually testing the change in this PR; looking at other runs, they seem to be having connectivity issues.  It's working for me locally:
```
$ make test-ci
# <snip>
/usr/bin/python tools/test.py  -p tap --logfile test.tap \
        --mode=release --flaky-tests=run \
         addons doctool known_issues message parallel sequential
1..1121
ok 1 parallel/test-arm-math-exp-regress-1376
  ---
  duration_ms: 0.222
  ...
ok 2 parallel/test-assert
  ---
  duration_ms: 0.342
  ...
```

gibm commented:
Makes sense, your change works manually for me as well.

bnoordhuis commented:
New CI: https://ci.nodejs.org/job/node-test-pull-request/2555/

jasnell commented:
LGTM

bnoordhuis commented:
I decided we could do a little better still for nested directories so PTAL.  Before, it printed:
```
ok 1 hello-world/test
```
Now it prints:
```
ok 1 addons/hello-world/test
```
CI: https://ci.nodejs.org/job/node-test-pull-request/2578/

bnoordhuis commented:
@jasnell Can I ask you to give this a quick look again?  The changes since the last review are in the second commit (not that the total diff is that huge: +9 −1.)

jasnell commented:
Still LGTM
